### DISCUSSION TITLE
transition for HmIP-BDT ?

### DISCUSSION THREAD

**Discussion by ChristophCaina** [SukramJ/aiohomematic#716] [Q&A]
Hi all,

I would like to setup a script or scene where the light that will be triggered by an HmIP-BDT Dimmable switch will slowly start to increase it's brightness from 0% to 100% and vice versa.

As far as I could see, this can be done by using "transition" in the automation or the script - but the device needs to support this.
Does anyone know, if this is possible to configure for that Dimmable switch?

I've recognized that there is a seperate sensor (level) on channel 3...
This does somehow reflect the current brightness-level but there, 100% will be shown as 1.0%
![grafik](https://user-images.githubusercontent.com/26391061/204337688-8def62c8-c33a-4ad8-bf8a-03753639ae0d.png)

Anything I can / need to setup within the CCU for this "transition"?

**Comment by SukramJ** [comment]
> Anything I can / need to setup within the CCU for this "transition"?

Not needed.

The sensor wil display the correct scale with the next release.

**Reply by ChristophCaina** [reply]
hey cool :)
So, in the next release transition will be supported?

**Reply by SukramJ** [reply]
Transition is supported from the beginning. Only the (not relevant) sensor scale will be updated in the next release.

**Reply by SukramJ** [reply]
Works/tested with HmIP-BDT, HmIP-FDT.

**Reply by ChristophCaina** [reply]
all right, thanks a lot :)
Then I will have a look, why it did not work the last time when I've tried it :)

**Reply by ChristophCaina** [reply]
thanks, I've tested the transition - and noticed some behave I don't understand...

While using the transition for turning the light on, it is working flawless and as expected.
But using transition for shutting the light off - it seems, that transition does not work.

it does ignore all time settings I've tried for the transition - the light will be switched off immediately.

Do you have any idea?

Here's the script that should do the transition to off:

```
alias: Licht GuteNacht
sequence:
  - service: light.turn_off
    data:
      transition: 200
    target:
      entity_id: light.dimmschalter_licht_schlafzimmer
mode: single
icon: mdi:ceiling-light-outline

```

**Reply by SukramJ** [reply]
These information are quiet useful and should have been part of your first post.

`transition` for `turn_off` is currently not implemented. I don't know if i have overseen this when imlementing `turn_on`, or if `transition` for `turn_off` is quiet new, but this requires a new version.

With `turn_on` you can dim down to 1% only.

**Reply by SukramJ** [reply]
Should work with 1.22.0

**Reply by ChristophCaina** [reply]
thanks :)
Honestly, when I tested it the first time, I didn't noticed that it behaves differently... after your feedback I did a few more steps trying to reproduce / understand what I did wrong the first time :)

Sorry for that confusion.
